### PR TITLE
Improving index page load speed

### PR DESCRIPTION
**Pull request by manangouhari**
- Add `defer` to script tags.
- Compress images

### CODE REVIEW

**Comment by bhaskarSingh** [comment]
Please go this https://developers.google.com/web/fundamentals/performance/optimizing-content-efficiency/loading-third-party-javascript#use_async_or_defer to understand the difference b/w defer & async.

This still doesn't solve the underlying issue stated in #220.

**Comment by manangouhari** [comment]
Update: Been trying to load the scripts dynamically, but it's ridden with bugs. Will spend tomorrow on this.

**Comment by manangouhari** [comment]
Not all scripts can be loaded async. 
The ones that can be, I'm loading them async.

**Comment by manangouhari** [comment]
It works now @bhaskarSingh . Ready to be merged.

File: gatsby-browser.js [hunk 1]
```diff
@@ -24,3 +25,48 @@ export const wrapRootElement = ({ element }) => {
...
+  script.onload = () => console.log('loaded: %s', url);
+  document.body.appendChild(script);
+};
+
+export const onRouteUpdate = ({ location, prevLocation }) => {
+  if (
+    location.pathname.indexOf('module-02') != -1 ||
+    location.pathname.indexOf('module-03') != -1
+  ) {
+    debugger;
```
Comment (bhaskarSingh) [inline gatsby-browser.js:44]: `debugger` I think this can be removed now.

File: gatsby-browser.js [hunk 2]
```diff
@@ -24,3 +25,48 @@ export const wrapRootElement = ({ element }) => {
...
+  const script = document.createElement('script');
+  script.src = url;
+  script.async = true;
+  script.type = 'text/javascript';
+  script.charSet = 'utf-8';
+  script.onload = () => console.log('loaded: %s', url);
+  document.body.appendChild(script);
+};
+
+export const onRouteUpdate = ({ location, prevLocation }) => {
```
Comment (bhaskarSingh) [inline gatsby-browser.js:39]: @manangouhari  Please add a comment within the `onRouteUpdate` function to elaborate on the logical breakdown of the code.

Comment (manangouhari) [reply gatsby-browser.js:38]: Done, take a look now.

**Comment by bhaskarSingh** [comment]
Screenshot 2021-03-10 at 1 08 54 AM
Throw above error when going throw module-3 chapter 1

**Comment by manangouhari** [comment]
> Screenshot 2021-03-10 at 1 08 54 AM
> Throws above error when going throw module-3 chapter 1

I'm not able to replicate it, can you please record a video and send.

**Comment by bhaskarSingh** [comment]
> I'm not able to replicate it, can you please record a video and send.

https://www.loom.com/share/be6ae0452efc40e6927657f7b719b215?from_recorder=1

Earlier it seems to be working i.e before the refactoring.

**Comment by manangouhari** [comment]
This is what I'm getting

image

**Comment by bhaskarSingh** [comment]
Try running it incognito.

**Comment by manangouhari** [comment]
Still can't replicate it. You take a look at it. 
In the meanwhile, let me know what I can take up next. 
One of them is #218

**Pull request merged**